Assets/ConnectApp/Screens/RecommendArticleScreen.cs

```diff
@@ -188,25 +190,30 @@ class _RecommendArticleScreenState : AutomaticKeepAliveClientMixin<RecommendArti
             );
         }
 
-        Widget _buildArticleList() {
+        Widget _buildArticleList(BuildContext context) {
             Widget content;
             var recommendArticleIds = this.widget.viewModel.recommendArticleIds;
-            if (this.widget.viewModel.articlesLoading && recommendArticleIds.isEmpty()) {
+            if (!this._hasBeenLoaded || this.widget.viewModel.articlesLoading && recommendArticleIds.isEmpty()) {
                 content = ListView.builder(
                     physics: new NeverScrollableScrollPhysics(),
                     itemCount: 6,
                     itemBuilder: (cxt, index) => new ArticleLoading()
                 );
+                this._hasBeenLoaded = true;
```

Comment:
To my understanding, _hasBeenLoaded is used to ensure that:  the first-build (and only the first-build) of this widget should go to this if (!this._hasBeenLoaded || XXXX) {} branch which shows a loading panel. However, i don't think it is strictly true that:  the second and later builds only happens after the data is eventually loaded. As the result, the data might still not be ready for displaying when this._hasBeenLoaded == true.

To address this issue,  is it possible to put this._hasBeenLoaded = true in some callback of the network responds, so that it can be called only after the data is "really" loaded ?

Reply:
you're right, I will try to modify that.